### ISSUE TITLE
"Hello world" neural network crashing

### ISSUE THREAD

**Issue by SanPen** [mlpack/mlpack#3447]
Hi, 

I'm trying to get a very simple NN example working and I get crashes most of the times.

I'm on Armadillo 12 and MLPACK 4.0.1. and VS2022 on windows 10. 

See the attached test data:
[X.csv.txt](https://github.com/mlpack/mlpack/files/11018240/X.csv.txt)
[Y.csv.txt](https://github.com/mlpack/mlpack/files/11018241/Y.csv.txt)

The example code is comming from the documentation and is this:

```
void main( ) {

        using namespace mlpack;
        // Initialize the network.
        FFN<> model;

        model.Add<Linear>(8);
        model.Add<Sigmoid>();
        model.Add<Linear>(3);
        model.Add<LogSoftMax>();

        arma::mat _X, _Y;
        _X.load("X.csv", arma::csv_ascii);
        _Y.load("Y.csv", arma::csv_ascii);

        std::cout << "X shape = " << _X.n_rows << ", " << _X.n_cols << "\n";
        std::cout << "Y shape = " << _Y.n_rows << ", " << _Y.n_cols << "\n";

        model.Train(_X, _Y);

}
```

I'm getting: `Process finished with exit code -1073741819 (0xC0000005)`

When I debug it leads to an array copy error in armadillo:

![imagen](https://user-images.githubusercontent.com/13423311/226353464-accd7a54-f966-40e4-8b3d-f65a50a14ef2.png)

**Comment by SanPen** [comment]
Hi, I get no extra information unfortunatelly.

The error is not reading files, but training.

**Comment by mrdaybird** [comment]
So, first of all you should use mlpack's data::Load method instead of armadillo's load methods. The difference is that mlpack's load method transposes the dataset upon load to fit its requirement of column-major data. 

Then, the other problem is with the dataset.  By default, FFN<> uses [NegativeLogLikelihood](https://www.mlpack.org/doc/mlpack-3.1.1/doxygen/classmlpack_1_1ann_1_1NegativeLogLikelihood.html) loss, which expects a class index(a single value) corresponding to each data point. The "Y.csv" file should contain series of rows containing a single value as label. 

Otherwise, if you are trying to achieve something else, explore some other loss functions, such as [CrossEntropyError](https://www.mlpack.org/doc/mlpack-3.1.1/doxygen/classmlpack_1_1ann_1_1CrossEntropyError.html) but in that case you have to make sure that the output dimensions corresponding to each data point is equal to number of classes.

**Comment by mrdaybird** [comment]
Well, I don't know the methods or model that can be used in your case, because it is very different from a simple regression or classification task. Sorry, but I hope you find a solution to your problem.
Also, I hope you got an answer for the original question about why your code was throwing a runtime error.

**Comment by aadi-raj** [comment]
> Alright!
> 
> Maybe it is because I'm too used to scipy's interfaces.
> 
> In general, what I want to achieve is a regression. So I train an artifact to recognize relations between X and Y, and later I want to predict any new X. In my case Y is always multidimensional.
> 
> What would be the right path for that?

I have dug the problem little bit. Below code is working fine for your problem statement.
`
//code
#include <mlpack.hpp>

using namespace std;
using namespace mlpack;

int main() {

        // Initialize the network.
        FFN<MeanSquaredError,HeInitialization> model;

        model.Add<Linear>(41);
        model.Add<ReLU>();
        model.Add<Linear>(26);
        arma::mat _X, _Y;
        data::Load("X.csv",_X);
        data::Load("Y.csv",_Y);

        std::cout << "X shape = " << _X.n_rows << ", " << _X.n_cols << "\n";
        std::cout << "Y shape = " << _Y.n_rows << ", " << _Y.n_cols << "\n";
        model.Train(_Y,_X);
        arma::vec row = {9.49,	0,	0,	0,	1.82,	1.2,	2.07,
       	2.35,	0.88,	0,	15.69,	0,	13.22,	18.07,	3.84,	3.05,
       	2.98,	4.39,	4.94,	0.07,	0,	2.5,	0,	3.65,	0,	0,
       	16.17,	16.47,	16.38,	12.17	,13.31,	0	,11.2	,0	,0	,
        0	,19.93	,12.77	,4.77	,4.36	,4.91};
        arma::mat ans;

        model.Predict(row,ans);
        cout<<ans<<endl;
}`

Let me know if it is not clear.
